zio-config/src/main/scala/zio/config/actions/ConfigDocs.scala

```diff
@@ -6,10 +6,16 @@ sealed trait ConfigDocs
 
 object ConfigDocs {
 
-  final case class Empty()                                                                      extends ConfigDocs
-  final case class PathDetails(path: Vector[String], value: Option[String], docs: List[String]) extends ConfigDocs
-  final case class And(left: ConfigDocs, right: ConfigDocs)                                     extends ConfigDocs
-  final case class Or(left: ConfigDocs, right: ConfigDocs)                                      extends ConfigDocs
+  final case class Empty() extends ConfigDocs
+  final case class PathDetails(
+    path: Vector[String],
+    value: Option[String],
+    source: Option[String],
+    docs: List[String]
+  ) extends ConfigDocs
+  final case class SourceDescription(sourceDescription: String, configDocs: ConfigDocs) extends ConfigDocs
```

Comment:
A bit confused here. How is this node conceptually different from the source parameter in PathDetails?

Reply:
We can have primary sources, like a scala map or system properties, or a source composed of multiple source (such as system properties with a scala map as a fallback). `SourceDescription` describes the source as a whole, for example, `"Scala Map - south details: with fallback source: Scala Map - east details (on all errors)"`. The source parameter in `PathDetails` tells you precisely where the value was obtained (ie, system properties, scala map or config descriptor).